tests/serac_odes.cpp

```diff
@@ -0,0 +1,645 @@
+  if (m == serac::TimestepMethod::SDIRK33) return 3;  // ?
+  if (m == serac::TimestepMethod::ForwardEuler) return 1;
+  if (m == serac::TimestepMethod::RK2) return 2;     // ?
+  if (m == serac::TimestepMethod::RK3SSP) return 3;  // ?
+  if (m == serac::TimestepMethod::RK4) return 4;
+  if (m == serac::TimestepMethod::GeneralizedAlpha) return 2;  // ?
+  if (m == serac::TimestepMethod::ImplicitMidpoint) return 2;
+  if (m == serac::TimestepMethod::SDIRK23) return 2;
+  if (m == serac::TimestepMethod::SDIRK34) return 3;
+
+  // for second order odes
+  if (m == serac::TimestepMethod::Newmark) return 2;
+  if (m == serac::TimestepMethod::HHTAlpha) return 2;
+  if (m == serac::TimestepMethod::WBZAlpha) return 2;             // ?
+  if (m == serac::TimestepMethod::AverageAcceleration) return 2;  // ?
+  if (m == serac::TimestepMethod::LinearAcceleration) return 2;   // ?
+  if (m == serac::TimestepMethod::CentralDifference) return 2;    // ?
+  if (m == serac::TimestepMethod::FoxGoodwin) return 2;           // ?
+
+  return -1;
+}
+
+int which_kind_of_ode(serac::TimestepMethod m)
+{
+  // for first order odes
+  if (m == serac::TimestepMethod::BackwardEuler) return 1;
+  if (m == serac::TimestepMethod::SDIRK33) return 1;
+  if (m == serac::TimestepMethod::ForwardEuler) return 1;
+  if (m == serac::TimestepMethod::RK2) return 1;
+  if (m == serac::TimestepMethod::RK3SSP) return 1;
+  if (m == serac::TimestepMethod::RK4) return 1;
+  if (m == serac::TimestepMethod::GeneralizedAlpha) return 1;
+  if (m == serac::TimestepMethod::ImplicitMidpoint) return 1;
+  if (m == serac::TimestepMethod::SDIRK23) return 1;
+  if (m == serac::TimestepMethod::SDIRK34) return 1;
+
+  // for second order odes
+  if (m == serac::TimestepMethod::Newmark) return 2;
+  if (m == serac::TimestepMethod::HHTAlpha) return 2;
+  if (m == serac::TimestepMethod::WBZAlpha) return 2;
+  if (m == serac::TimestepMethod::AverageAcceleration) return 2;
+  if (m == serac::TimestepMethod::LinearAcceleration) return 2;
+  if (m == serac::TimestepMethod::CentralDifference) return 2;
+  if (m == serac::TimestepMethod::FoxGoodwin) return 2;
+
+  return -1;
+}
+
+double first_order_ode_test(int nsteps, ode_type type, constraint_type constraint, TimestepMethod timestepper,
+                            DirichletEnforcementMethod enforcement)
+{
+  double t           = 0.0;
+  double dt          = 1.0 / nsteps;
+  double previous_dt = -1.0;
+  double c0;
+
+  mfem::Vector x(3);
+  mfem::Vector previous(3);
+  previous = 0.0;
+
+  mfem::DenseMatrix J(3, 3);
+
+  mfem::Mesh               mesh1D(2);
+  mfem::ParMesh            mesh(MPI_COMM_WORLD, mesh1D);
+  BoundaryConditionManager bcs(mesh);
+
+  // although these test problems don't really apply to a finite element mesh,
+  // the tools in serac require that these finite element data structures exist
+  serac::FiniteElementState dummy(mesh, FiniteElementState::Options{.order = 1, .name = "dummy"});
```

Comment:
I believe this defaults to a vector valued FEstate. Is that what you want? I am reading this correctly @joshessman-llnl ?

Reply:
Also I wonder if we should default to a scalar field.

Reply:
I intended to create a scalar-valued FEstate. I would assume omitting a number of components per node would imply scalar-valued.

Reply:
It defaults to the dimension of the mesh, which in this case is 1.

Reply:
I'll make another PR to change this default. I think it's a different behavior than most FEM libraries so it could be confusing.